beacon_node/beacon_chain/src/data_availability_checker/availability_view.rs

```diff
@@ -198,7 +207,9 @@ pub trait AvailabilityView<E: EthSpec> {
     /// of expected blobs.
     fn is_available(&self) -> bool {
         if let Some(num_expected_blobs) = self.num_expected_blobs() {
+            // Note: the first equality covers the case of no columns to sample post EIP-7594
             num_expected_blobs == self.num_received_blobs()
+                || self.num_received_columns() >= self.samples_per_slot()
```

Comment:
Note that the `get_cached_data_columns` currently stores the gossip verified data columns, so it's really the `custody_columns` rather than the sample columns.